File: Lab6/app/src/main/java/com/example/lab6/MainActivity.java
package com.example.lab6;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.BaseAdapter;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.Spinner;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    class Data{
        int photo;
        String name;
    }
    public class MyAdapter extends BaseAdapter{
        private MainActivity.Data[] data;
        private int view;
        public MyAdapter(MainActivity.Data[] data,int view){
            this.data= data;
            this.view= view;
        }
        @Override
        public int getCount(){
            return data.length;
        }
        @Override
        public Object getItem(int position){
            return data[position];
        }
        @Override
        public long getItemId(int position){
            return 0;
        }
        @Override
        public View getView(int position, View converView, ViewGroup parent){
            converView= getLayoutInflater().inflate(view,parent,false);
            TextView name = converView.findViewById(R.id.name);
            name.setText(data[position].name);
            ImageView imageView = converView.findViewById(R.id.imageView);
            imageView.setImageResource(data[position].photo);
            return converView;
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        String[] transNameArray= new String[]{"腳踏車","機車","汽車","巴士"};
        int[] transPhotoIdArray = new int[]{R.drawable.trans1,R.drawable.trans2,R.drawable.trans3,R.drawable.trans4};
        Data[] transData =new Data[transNameArray.length];
        for(int i=0;i<transData.length;i++){
           transData[i]=new Data();
           transData[i].name=transNameArray[i];
           transData[i].photo=transPhotoIdArray[i];
        }

        MyAdapter transAdapter = new MyAdapter(transData,R.layout.trans_list);
        Spinner spinner=findViewById(R.id.spinner);
        spinner.setAdapter(transAdapter);

        String[] messageArray = new String[]{"訊息1","訊息2","訊息3","訊息4","訊息5","訊息6"};
        ArrayAdapter<String> messageAdapter =new ArrayAdapter<>(this, android.R.layout.simple_list_item_1,messageArray);
        ListView listview= findViewById(R.id.listView);
        listview.setAdapter(messageAdapter);

        String[] cubeeNameArray= new String[]{"哭哭","發抖","再見","生氣","昏倒","竊笑","很棒","你好","驚嚇","大笑"};
        int[] cubeePhotoArray = new int[]{R.drawable.cubee1,R.drawable.cubee2,R.drawable.cubee3,R.drawable.cubee4
                ,R.drawable.cubee5,R.drawable.cubee6,R.drawable.cubee7,R.drawable.cubee8,R.drawable.cubee9,R.drawable.cubee10};
        Data[] cubeeData= new Data[cubeeNameArray.length];
        for(int i=0;i<cubeeData.length;i++){
            cubeeData[i]=new Data();
            cubeeData[i].name=cubeeNameArray[i];
            cubeeData[i].photo=cubeePhotoArray[i];
        }
        MyAdapter cubeeAdapter =new MyAdapter(cubeeData,R.layout.cubee_list);
        GridView gridView=findViewById(R.id.gridView);
        gridView.setAdapter(cubeeAdapter);
        gridView.setNumColumns(3);
    }
}

File: Lab6_withKotlin/app/src/main/java/com/stella/lab6_withkotlin/MainActivity.kt
package com.stella.lab6_withkotlin

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    inner class Data {
        var photo = 0
        var name: String? = null
    }

    inner class MyAdapter(private val data: Array<Data>, private val view: Int) :
        BaseAdapter() {
        override fun getCount(): Int {
            return data.size
        }

        override fun getItem(position: Int): Any {
            return data[position]
        }

        override fun getItemId(position: Int): Long {
            return 0
        }

        override fun getView(position: Int, converView: View, parent: ViewGroup): View {
            var converView = converView
            converView = layoutInflater.inflate(view, parent, false)
            val name = converView.findViewById<TextView>(R.id.name)
            name.text = data[position].name
            val imageView = converView.findViewById<ImageView>(R.id.imageView)
            imageView.setImageResource(data[position].photo)
            return converView
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val transNameArray = arrayOf("腳踏車", "機車", "汽車", "巴士")
        val transPhotoIdArray =
            intArrayOf(R.drawable.trans1, R.drawable.trans2, R.drawable.trans3, R.drawable.trans4)
        val transData = arrayOfNulls<Data>(transNameArray.size)
        for (i in transData.indices) {
            transData[i] = Data()
            transData[i]!!.name = transNameArray[i]
            transData[i]!!.photo = transPhotoIdArray[i]
        }
        val transAdapter: MyAdapter = MyAdapter(transData, R.layout.trans_list)
        val spinner = findViewById<Spinner>(R.id.spinner)
        spinner.adapter = transAdapter
        val messageArray = arrayOf("訊息1", "訊息2", "訊息3", "訊息4", "訊息5", "訊息6")
        val messageAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, messageArray)
        val listview = findViewById<ListView>(R.id.listView)
        listview.adapter = messageAdapter
        val cubeeNameArray = arrayOf("哭哭", "發抖", "再見", "生氣", "昏倒", "竊笑", "很棒", "你好", "驚嚇", "大笑")
        val cubeePhotoArray = intArrayOf(
            R.drawable.cubee1,
            R.drawable.cubee2,
            R.drawable.cubee3,
            R.drawable.cubee4,
            R.drawable.cubee5,
            R.drawable.cubee6,
            R.drawable.cubee7,
            R.drawable.cubee8,
            R.drawable.cubee9,
            R.drawable.cubee10
        )
        val cubeeData = arrayOfNulls<Data>(cubeeNameArray.size)
        for (i in cubeeData.indices) {
            cubeeData[i] = Data()
            cubeeData[i]!!.name = cubeeNameArray[i]
            cubeeData[i]!!.photo = cubeePhotoArray[i]
        }
        val cubeeAdapter: MyAdapter = MyAdapter(cubeeData, R.layout.cubee_list)
        val gridView = findViewById<GridView>(R.id.gridView)
        gridView.adapter = cubeeAdapter
        gridView.numColumns = 3
    }
}